server/lib/src/entry.rs

```diff
@@ -479,6 +488,11 @@ impl Entry<EntryInit, EntryNew> {
         self.attrs.remove(attr);
     }
 
+    /// Set the content of this ava with this valueset, ignoring the previous data.
+    pub fn set_ava_set(&mut self, attr: &Attribute, vs: ValueSet) {
+        self.attrs.insert(attr.clone(), vs);
```

Comment:
Doing an insert on a BTreeMap adds to the values, it doesn't replace them...

Reply:
https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#method.insert

"If the map did have this key present, the value is updated, and the old value is returned. "

It does replace them. The difference to "replace" is that it won't add if not present. So this code is correct.

Reply:
oh right, it's just updating that one key, ffs brain.